src/Money.php

```diff
@@ -62,35 +84,28 @@ final class Money implements \JsonSerializable
     /**
      * @param int|string $amount Amount, expressed in the smallest units of $currency (eg cents)
      *
-     * @throws \InvalidArgumentException If amount is not integer
+     * @throws InvalidArgumentException If amount is not integer
      */
     public function __construct($amount, Currency $currency)
     {
+        $this->currency = $currency;
+
+        if (is_int($amount)) {
```

Comment:
Unsure about this one: the added function call seems to bring little, and perhaps even make things worse than `filter_var()` :-\

Reply:
Then let's drop it.